server/store/datastore/pipeline.go

```diff
@@ -52,9 +52,18 @@ func (s storage) GetPipelineLastBefore(repo *model.Repo, branch string, num int6
 		Get(pipeline))
 }
 
-func (s storage) GetPipelineList(repo *model.Repo, p *model.ListOptions) ([]*model.Pipeline, error) {
+func (s storage) GetPipelineList(repo *model.Repo, p *model.ListOptions, f *model.FilterOptions) ([]*model.Pipeline, error) {
 	pipelines := make([]*model.Pipeline, 0, 16)
-	return pipelines, s.paginate(p).Where("pipeline_repo_id = ?", repo.ID).
+
+	if f != nil {
+		return pipelines, s.paginate(p).Where(builder.Eq{"pipeline_repo_id": repo.ID}).
```

Comment:
i would create a new condition with builder and set `builder.Eq{"pipeline_repo_id": repo.ID}` on the beginning

then have an if that will enable filters based on options:
```go
f.After != 0 
  cond = cond.And(builder.Gt{"pipeline_started": f.After}
}
```
...

Reply:
PS: defining an filter type to pass into the store func is the right way yes :)

Reply:
Thanks!